lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemModule.java

```diff
@@ -104,9 +108,14 @@ protected void setup(Binder binder)
         newOptionalBinder(binder, CacheKeyProvider.class).setDefault().to(DefaultCacheKeyProvider.class).in(Scopes.SINGLETON);
 
         newOptionalBinder(binder, TrinoFileSystemCache.class);
+        newOptionalBinder(binder, MemoryFileSystemCache.class);
 
+        boolean isCoordinator = nodeManager.getCurrentNode().isCoordinator();
         if (config.isCacheEnabled()) {
-            install(new AlluxioFileSystemCacheModule(nodeManager.getCurrentNode().isCoordinator()));
+            install(new AlluxioFileSystemCacheModule(isCoordinator));
+        }
+        if (enableCoordinatorFilesCache) {
```

Comment:
`iceberg.metadata-cache.enabled` is true on non-coordinators too. Why not install `MemoryFileSystemCacheModule` only when `isCoordinator==true`?

Reply:
Check the implementation of MemoryFileSystemCacheModule again, the cache is installed only on coordinator.
The module used to bind MemoryFileSystemCacheConfig on all nodes so that users are not forced to split their catalog configuration for coordinator and worker.